source/data-formats/modeling-bson-data.txt

```diff
@@ -11,140 +11,198 @@ Modeling BSON Data
+{+language+} values, at the following levels:
+
+- ``MongoDB\Client``, which sets the *default for all operations* unless overridden
+- ``MongoDB\Database``
+- ``MongoDB\Collection``
+
+This list also indicates the increasing order of precedence of the option settings. For
+example, if you set a ``typeMap`` for a collection, it will override the type map
+set on the database.
+
+The {+library-short+} uses the following type map by default:
 
 .. code-block:: php
 
-   <?php
+   [
+       'array' => 'MongoDB\Model\BSONArray',
+       'document' => 'MongoDB\Model\BSONDocument',
+       'root' => 'MongoDB\Model\BSONDocument',
+   ]
 
-   $collection = (new MongoDB\Client)->test->persons;
+This type map performs the following conversions:
 
-   $result = $collection->insertOne(new Person('Bob'));
+- Arrays to ``MongoDB\Model\BSONArray`` objects
+- Top-level and embedded BSON documents to ``MongoDB\Model\BSONDocument`` objects
 
-   $person = $collection->findOne(['_id' => $result->getInsertedId()]);
+A type map can specify any class that implements
+:php:`MongoDB\BSON\Unserializable <mongodb-bson-unserializable>`. It can
+also specify the ``array``, ``stdClass``, and ``object`` types.
 
-   var_dump($person);
+Custom Type Map Example
+```````````````````````
 
-The output would then resemble:
+The following example sets the ``typeMap`` option for the ``restaurants`` collection
+that serializes arrays and BSON documents as ``MongoDB\Model\BSONDocument`` objects:
 
-.. code-block:: none
+.. literalinclude:: /includes/bson.php
+    :language: php
+    :dedent:
+    :start-after: start-type-map
+    :end-before: end-type-map
 
-   object(Person)#18 (3) {
-     ["id":"Person":private]=>
-     object(MongoDB\BSON\ObjectId)#15 (1) {
-       ["oid"]=>
-       string(24) "56fad2c36118fd2e9820cfc1"
-     }
-     ["name":"Person":private]=>
-     string(3) "Bob"
-     ["createdAt":"Person":private]=>
-     object(MongoDB\BSON\UTCDateTime)#17 (1) {
-       ["milliseconds"]=>
-       int(1459278531218)
-     }
-   }
+.. _php-persistable-classes:
+
+Persistable Classes
+~~~~~~~~~~~~~~~~~~~
+
+You can create classes that implement the ``MongoDB\BSON\Persistable`` interface,
+which instructs the {+library-short+} to automatically perform serialization and
+deserialization according to the {+extension-short+}'s :php:`persistence specification
+<mongodb.persistence>`. The ``Persistable`` interface is analogous to {+language+}'s
+:php:`Serializable interface <class.serializable>`.
+
+When deserializing a PHP variable from BSON, the encoded class name of a
+``Persistable`` object overrides any class specified in the ``typeMap`` option.
+However, it does not override ``array``,  ``stdClass``, or ``object`` types.
 
-The same document in the MongoDB shell might display as:
+.. tip::
+
+   To learn more about the ``Persistable`` interface, see :php:`Persistable <mongodb-bson-persistable>`
+   in the {+extension-short+} documentation.
```

Comment:
S: Instead of adding this admonition, inline the link when you first mention `Persistable`